src/datumaro/plugins/data_formats/datumaro/importer.py

```diff
@@ -28,9 +28,11 @@ def detect(
         with context.probe_text_file(
             annot_file,
             'must be a JSON object with "categories" ' 'and "items" keys',
-        ) as f:
-            contents = parse_json(f.read())
-            if not {"categories", "items"} <= contents.keys():
+        ):
+            fpath = osp.join(context.root_path, annot_file)
```

Comment:
Here, I personally love to get access to context.root_path directly, but this is recommended to avoid during detection https://github.com/openvinotoolkit/datumaro/blob/4faaae505b849f3d280d13366fded37e9610afb6/src/datumaro/components/format_detection.py#L178-L179. Do you have any objection to use this during detection

Reply:
That guide is because the methods provided by `context` are using `self._root_path` internally. For example,
https://github.com/openvinotoolkit/datumaro/blob/4faaae505b849f3d280d13366fded37e9610afb6/src/datumaro/components/format_detection.py#L354-L357

However, `JsonSectionPageMapper` Rust API has no information for `root_path`. There is no choice but to inject the full path `osp.join(context.root_path, annot_file)` to access the file.

Reply:
Agree, how about other formats with pure Python codes? I just want to hear your opinion. 
Frankly speaking, I don't want to avoid the use of `context.root_path`.